### ISSUE TITLE
Error building native extension during installing inside Docker image on Mac M1

### ISSUE THREAD

**Issue by EvgeniyRRU** [ruby/racc#190]
Hello everyone. I've tried to build my Rails 7 application inside Docker on Mac M1 machine. At first I've seen this error:
```
An error occurred while installing racc (1.6.0), and Bundler cannot continue.
                                                             
In Gemfile:                                         
  rails was resolved to 7.0.1, which depends on
    actioncable was resolved to 7.0.1, which depends on                                                                   
      actionpack was resolved to 7.0.1, which depends on                                                                  
        actionview was resolved to 7.0.1, which depends on                                                                
          rails-dom-testing was resolved to 2.0.3, which depends on                                                       
            nokogiri was resolved to 1.13.1, which depends on                                                             
              racc                                            
```
I've tried to dig inside error output and faced with that output:

```
current directory: /usr/local/bundle/gems/racc-1.6.0/ext/racc/cparse
/usr/local/bin/ruby -I /usr/local/lib/ruby/3.0.0 -r ./siteconf20220519-9-vd84s0.rb extconf.rb
checking for rb_block_call()... no
checking for rb_ary_subseq()... no
creating Makefile

current directory: /usr/local/bundle/gems/racc-1.6.0/ext/racc/cparse
make DESTDIR\= clean

current directory: /usr/local/bundle/gems/racc-1.6.0/ext/racc/cparse
make DESTDIR\=
compiling cparse.c
during GIMPLE pass: pre
cparse.c: In function 'cparse_params_memsize':
cparse.c:429:1: internal compiler error: Segmentation fault
  429 | cparse_params_memsize(const void *ptr)
      | ^~~~~~~~~~~~~~~~~~~~~
mmap: Permission denied
Please submit a full bug report,
with preprocessed source if appropriate.
See <https://gcc.gnu.org/bugs/> for instructions.
make: *** [Makefile:245: cparse.o] Error 1

make failed, exit code 2
```
Nothing special inside my Dockerfile:
```dockerfile
FROM ruby:3.0.4-alpine
RUN apk add --no-cache \
  tzdata \
  build-base \
  ruby-dev \
  git \
  screen \
  bash \
  less \
  postgresql-dev

RUN wget -O - https://github.com/jemalloc/jemalloc/releases/download/5.2.1/jemalloc-5.2.1.tar.bz2 | tar xj && \
    cd jemalloc-5.2.1 && \
    ./configure && \
    make && \
    make install
ENV LD_PRELOAD /usr/local/lib/libjemalloc.so.2

RUN gem install bundler
ENV APP /app
ENV PATH $APP/bin:$PATH
RUN mkdir -p $APP
WORKDIR $APP
COPY Gemfile ./
COPY Gemfile.lock ./
RUN bundle install --without development test
COPY . .
```

As far as I know, that image builds fine on Mac machines with Intel chip.

Can you help me? Did you faced with same strange behavior?

**Comment by hsbt** [comment]
Is it reproduce without alpine image? We didn't support alpine especially `muslc`.

**Comment by EvgeniyRRU** [comment]
It just freezing 😞 

New Dockerfile:
```dockerfile
FROM ruby:3.0.4-slim

RUN apt-get update -qq \
  && DEBIAN_FRONTEND=noninteractive apt-get install -yq --no-install-recommends \
    build-essential \
    git \
    less \
    gnupg2 \
    wget \
    curl \
    libpq-dev \
    postgresql-client \
  && apt-get clean \
  && rm -rf /var/cache/apt/archives/* \
  && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* \
  && truncate -s 0 /var/log/*log

RUN wget -O - https://github.com/jemalloc/jemalloc/releases/download/5.2.1/jemalloc-5.2.1.tar.bz2 | tar xj && \
    cd jemalloc-5.2.1 && \
    ./configure && \
    make && \
    make install
ENV LD_PRELOAD /usr/local/lib/libjemalloc.so.2

RUN gem install bundler
ENV APP /app
ENV PATH $APP/bin:$PATH
RUN mkdir -p $APP
WORKDIR $APP
COPY Gemfile ./
COPY Gemfile.lock ./
# RUN bundle install --without development test
COPY . .
```

```
root@4e030344245c:/app# 
root@4e030344245c:/app# gem install racc -V -v=1.6.0
HEAD https://index.rubygems.org/
200 OK
GET https://index.rubygems.org/info/racc
200 OK
GET https://index.rubygems.org/quick/Marshal.4.8/racc-1.6.0.gemspec.rz
200 OK
Downloading gem racc-1.6.0.gem
GET https://index.rubygems.org/gems/racc-1.6.0.gem
Fetching racc-1.6.0.gem
200 OK
/usr/local/bundle/gems/racc-1.6.0/COPYING
/usr/local/bundle/gems/racc-1.6.0/ChangeLog
/usr/local/bundle/gems/racc-1.6.0/README.ja.rdoc
/usr/local/bundle/gems/racc-1.6.0/README.rdoc
/usr/local/bundle/gems/racc-1.6.0/TODO
/usr/local/bundle/gems/racc-1.6.0/bin/racc
/usr/local/bundle/gems/racc-1.6.0/doc/en/NEWS.en.rdoc
/usr/local/bundle/gems/racc-1.6.0/doc/en/grammar.en.rdoc
/usr/local/bundle/gems/racc-1.6.0/doc/en/grammar2.en.rdoc
/usr/local/bundle/gems/racc-1.6.0/doc/ja/NEWS.ja.rdoc
/usr/local/bundle/gems/racc-1.6.0/doc/ja/command.ja.html
/usr/local/bundle/gems/racc-1.6.0/doc/ja/debug.ja.rdoc
/usr/local/bundle/gems/racc-1.6.0/doc/ja/grammar.ja.rdoc
/usr/local/bundle/gems/racc-1.6.0/doc/ja/index.ja.html
/usr/local/bundle/gems/racc-1.6.0/doc/ja/parser.ja.rdoc
/usr/local/bundle/gems/racc-1.6.0/doc/ja/usage.ja.html
/usr/local/bundle/gems/racc-1.6.0/ext/racc/MANIFEST
/usr/local/bundle/gems/racc-1.6.0/ext/racc/cparse/cparse.c
/usr/local/bundle/gems/racc-1.6.0/ext/racc/cparse/extconf.rb
/usr/local/bundle/gems/racc-1.6.0/lib/racc.rb
/usr/local/bundle/gems/racc-1.6.0/lib/racc/compat.rb
/usr/local/bundle/gems/racc-1.6.0/lib/racc/debugflags.rb
/usr/local/bundle/gems/racc-1.6.0/lib/racc/exception.rb
/usr/local/bundle/gems/racc-1.6.0/lib/racc/grammar.rb
/usr/local/bundle/gems/racc-1.6.0/lib/racc/grammarfileparser.rb
/usr/local/bundle/gems/racc-1.6.0/lib/racc/info.rb
/usr/local/bundle/gems/racc-1.6.0/lib/racc/iset.rb
/usr/local/bundle/gems/racc-1.6.0/lib/racc/logfilegenerator.rb
/usr/local/bundle/gems/racc-1.6.0/lib/racc/parser-text.rb
/usr/local/bundle/gems/racc-1.6.0/lib/racc/parser.rb
/usr/local/bundle/gems/racc-1.6.0/lib/racc/parserfilegenerator.rb
/usr/local/bundle/gems/racc-1.6.0/lib/racc/sourcetext.rb
/usr/local/bundle/gems/racc-1.6.0/lib/racc/state.rb
/usr/local/bundle/gems/racc-1.6.0/lib/racc/statetransitiontable.rb
/usr/local/bundle/gems/racc-1.6.0/lib/racc/static.rb
Building native extensions. This could take a while...
current directory: /usr/local/bundle/gems/racc-1.6.0/ext/racc/cparse
["/usr/local/bin/ruby", "-I", "/usr/local/lib/ruby/3.0.0", "-r", "./siteconf20220519-10-90ttrn.rb", "extconf.rb"]
```

**Comment by EvgeniyRRU** [comment]
Upd. This problem happens with any ruby native extensions, not only **racc**.

Solved by moving to `ruby-3.0.4` base image.
